src/plugin/persist.js

```diff
@@ -158,14 +158,21 @@ eg.module("persist", ["jQuery", eg, window, document], function($, ns, global, d
 
 	function setStateByKey(key, data) {
 		var beforeData = getState();
-		beforeData[key] = data;
+
+		if (data === null || typeof data === "undefined" || data === "") {
```

Comment:
The only way to be called setStateByKey() is when data has truthy value.
It can be reached this condition?
```js
$.persist = function(state) {
	var key;
	var data;
	if (typeof state === "string") {
		key = state;
		data = arguments.length === 2 ? arguments[1] : null;
	} else {
		key = GLOBAL_KEY;
		data = arguments.length === 1 ? state : null;
	}
	data && setStateByKey(key, data);
	return getStateByKey(key);
};
```